### PR TITLE
react-avatar: Creating AvatarGroupItem component

### PR DESCRIPTION
**Pull request by sopranopillow**
This PR creates the `AvatarGroupItem` component that will be used by AvatarGroup.

## Related Issue(s)

#22240 and #22736

### CODE REVIEW

File: packages/react-components/react-avatar/config/api-extractor.local.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
   "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
   "extends": "./api-extractor.json",
-  "mainEntryPointFilePath": "<projectFolder>/dist/packages/react-components/<unscopedPackageName>/src/index.d.ts"
+  "mainEntryPointFilePath": "<projectFolder>/dist/types/packages/react-components/<unscopedPackageName>/src/index.d.ts"
```
Comment (behowell) [inline packages/react-components/react-avatar/config/api-extractor.local.json:4]: Out of curiosity, why is this change needed in this PR?

**Review by behowell — APPROVED** [review]

File: packages/react-components/react-avatar/config/api-extractor.local.json (hunk 1 above)
Comment (sopranopillow) [reply packages/react-components/react-avatar/config/api-extractor.local.json:4]: `create-component` added it, it doesn't seem to affect anything though. I can remove it just in case?

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/Avatar.fixture.js'>Avatar</abbr>                                                                                    |  `45.571 kB`<br />`13.318 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `181.141 kB`<br />`50.598 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `34.078 kB`<br />`11.098 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/87f555dfcbe007e701f80b56e3c69c50141e5466'>87f555dfcbe007e701f80b56e3c69c50141e5466</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23287/perf-test-react-components/Avatar-mount_base.html">934</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23287/perf-test-react-components/Avatar-mount.html">884</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23287/perf-test-react-components/Button-mount_base.html">567</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23287/perf-test-react-components/Button-mount.html">544</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23287/perf-test-react-components/FluentProvider-mount_base.html">1842</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23287/perf-test-react-components/FluentProvider-mount.html">1827</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23287/perf-test-react-components/FluentProviderWithTheme-mount_base.html">286</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23287/perf-test-react-components/FluentProviderWithTheme-mount.html">294</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23287/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">245</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23287/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">251</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23287/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">297</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23287/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">299</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23287/perf-test-react-components/MakeStyles-mount_base.html">1575</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23287/perf-test-react-components/MakeStyles-mount.html">1568</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-avatar/config/api-extractor.local.json (hunk 1 above)
Comment (behowell) [reply packages/react-components/react-avatar/config/api-extractor.local.json:4]: I'm not really sure what this actually does. But if it was added by create-component, and it doesn't break things, then it should be fine.
